src/illuminatio/test_orchestrator.py

```diff
@@ -207,7 +217,55 @@ def _rewrite_ports_for_host(self, port_list, services_for_host):
                 )
             else:
                 # TODO change to exception, handle it higher up
+                self.logger.error(
+                    "No service port found for Host: %s Port: %s. Available services %s",
+                    host_string,
+                    port,
+                    service_ports,
+                )
                 rewritten_ports[port] = "err"
+
+        for named_port in named_ports:
+            host_string_separated = str(host_string).split(":")
+            namespace = host_string_separated[0]
+            label_selector_string = (
+                host_string_separated[1] if host_string_separated[1] != "*" else ""
+            )
+
+            # Strip of the "-" and add it again after name has been resolved
+            prefix = "-" if "-" in named_port else ""
+            named_port_without_prefix = named_port.replace("-", "")
+
+            # We use a simple cache to avoid looking up the same pod-port combination twice for another test case
+            key_hostname_port = f"{host_string}_{named_port}"
+            if key_hostname_port not in self.portname_cache:
+                self.portname_cache[key_hostname_port] = resolve_port_name(
+                    namespace, label_selector_string, named_port_without_prefix
+                )
+
+            numerical_port_without_prefix = self.portname_cache[key_hostname_port]
+
+            service_ports_for_port = [
+                p
+                for p in service_ports
+                if p.target_port == numerical_port_without_prefix
+                or p.target_port == named_port_without_prefix
+            ]
+            if service_ports_for_port:
+                rewritten_ports[named_port] = "%s%s" % (
+                    prefix,
+                    str(service_ports_for_port[0].port),
+                )
+            else:
+                # TODO change to exception (explanation see above)
```

Comment:
which is the explanation?

Reply:
The same as for the numerical ports. Unfortunately the comment is not within the "review window":

"# TODO this was a hotfix for recipe 11, where ports 53 were allowed but not for any target,
            # resulting in test to 53 being written despite no service matching them existing.
            # That error should be handled in test generation, an exception here would be fine"